components/ide/jetbrains/cli/cmd/root.go

```diff
@@ -21,4 +25,16 @@ func Execute() {
 	}
 }
 
+func getCliApiUrl() *url.URL {
+	var backendPort = 63342
+	if _, fileStatError := os.Stat("/ide-desktop/bin/idea-cli-dev"); !errors.Is(fileStatError, os.ErrNotExist) {
+		backendPort = backendPort + 1
```

Comment:
What's this used for? If we make `backendPort` in func, it can only be `63342` or `63343`

Reply:
`63342` is the default port used by JetBrains "remote-dev-server.sh".

For developing the `components/ide/jetbrains/backend-plugin`, we need to use IntelliJ IDEA on Gitpod. Which means we'll be connecting to the remote-dev-server that is running in `63342`.

For testing changes on backend-plugin, we run `launch-dev-server.sh`.
This starts a new remote-dev-server on that workspace [[1](https://github.com/gitpod-io/gitpod/blob/8a1930a9ea5eeb955bb3ae0564f9e53e1a9d1c98/components/ide/jetbrains/backend-plugin/README.md#development)].
This new remote-dev-server will look for the availability of port `63342`, and if it's not available, it will increment the port by 1 until it finds an open port.
So we can say it's always going to pick `63343`.

Also note that `/ide-desktop/bin/idea-cli-dev` is created by `launch-dev-server.sh` right before launching the remote-dev-server. That's why we're using port `63343` if that file exists.

Reply:
Let me update the code with the addition of this comment explaining the magic number.

Thanks for asking!